### PR TITLE
Copy file permissions

### PR DESCRIPTION
**Pull request by djaglowski**
The `Copy` function now copies permissions from the source file to the destination file.

Addresses #1

### CODE REVIEW

**Comment by hugocarreira** [comment]
Nice @djaglowski !!
I will give a validated and already accepted! 

thank u 😄

File: decent_copy.go [hunk 1]
```diff
@@ -16,7 +21,7 @@ func Copy(filepathOrigin, filepathDestiny string) error {
 
 	io.Copy(destFile, srcFile)
 
-	err := destFile.Sync()
+	_ = os.Chmod(filepathDestiny, srcInfo.Mode())
```
Comment (hugocarreira) [inline decent_copy.go]: I think here you can take the error and validate it.. 
What do you think?

Comment (djaglowski) [reply decent_copy.go]: I agree, but we will either need a test that generates the error, or we need to configure Codecov to not enforce 100% coverage.

I will make the change and then look into whether I can generate/mock the error in tests.

Comment (djaglowski) [reply decent_copy.go]: I believe that one of these has to happen in order to test this:
Option 1: Mock `os.Chmod`
Option 2: Make `os.Chmod` fail immediately _after_ we have just created `filepathDestiny`

I'm not sure how to do either of these. What do you think?

Comment (djaglowski) [reply decent_copy.go]: Looks like a similar issue was encountered with `io.Copy`, and the decision there was to remove the error handling altogether.

In my opinion, it is more correct to return the error in both cases, even if test coverage does not validate the `if err != nil { return err }` part.

Comment (djaglowski) [reply decent_copy.go]: Ok, I found this library: https://github.com/dave/courtney

It appears to have been created for exactly this situation. Basically, it just excludes simple error return statements from code coverage.

I added it to this PR and it seems to work as intended. Let me know what you think.

Comment (djaglowski) [reply decent_copy.go]: Hi @hugocarreira, I believe this is ready for a follow up review.

**Comment by hugocarreira** [comment]
@djaglowski merged!

thanksss @djaglowski
